[Security Solution] Fix issue where Endpoint list page crashes on missing Policy Id

kevinlog opened the pull request:
## Summary

This PR addresses a potential issue seen in at least one customer environment where the Elastic Endpoint streams an Endpoint document with blank Policy information, which causes the Endpoint list page to crash.  This Endpoint issue is being addressed separately in the Endpoint.

However, on the Kibana side, we want to keep the Endpoint list page from crashing and becoming unusable in the event that this happens with older Endpoints.  We also mark that the policy information is not found.

Before:
<img width="1728" alt="image" src="https://user-images.githubusercontent.com/56395104/182243365-b3354bf6-68e0-42a4-9373-ab6516f867e3.png">

After:
<img width="1728" alt="image" src="https://user-images.githubusercontent.com/56395104/182243196-6571d78d-0c89-43d9-916f-9f6351630eb4.png">


### Checklist

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)

elasticmachine commented:
Pinging @elastic/security-onboarding-and-lifecycle-mgt (Team:Onboarding and Lifecycle Mgt)

paul-tavares commented on x-pack/plugins/security_solution/public/management/components/endpoint_policy_link.tsx:
```diff
@@ -25,20 +26,32 @@ export const EndpointPolicyLink = memo<
     backLink?: PolicyDetailsRouteState['backLink'];
   }
 >(({ policyId, backLink, children, missingPolicies = {}, ...otherProps }) => {
+  const isPolicyIdMissing = policyId === undefined || policyId === '';
```
can you just `!policyId`? just to make sure we don't miss a non-string case here like `null` or `0`.

paul-tavares commented on x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/utils.ts:
```diff
@@ -13,7 +13,12 @@ export const isPolicyOutOfDate = (
 ): boolean => {
-  if (current === undefined || current === null) {
+  if (
+    current === undefined ||
+    current === null ||
+    reported.id === undefined ||
+    reported.id === ''
```
replace with ` !current || !reported.id`

paul-tavares reviewed (commented):
Left some feedback, but minor stuff.

Also - can you test Responder `status` command on a host that has no policy id? I know its not a link, but wondering if we have an issue there as well when the value is not set (or any other place we show the link to the policy details

kevinlog commented:
@paul-tavares addressed your comments.

I also tested with the Responder and the status command.  Nothing crashes, we just leave the Policy name blank.  I think that's OK since if the user gets into this state, they likely need to look at the Agent/Endpoint installation.  If Endpoint isn't sending Policy info, something deeper is wrong.

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/56395104/182470075-b2c1b384-b3f4-4033-8861-c1843701ab43.png">

paul-tavares reviewed (approved):
🚢 it

kevinlog commented:
@elasticmachine merge upstream

kevinlog commented:
@elasticmachine merge upstream

kevinlog commented:
@elasticmachine merge upstream

kevinlog commented:
@elasticmachine merge upstream
